src/v/cluster/controller.h

```diff
@@ -159,6 +159,10 @@ class controller {
         return _raft0->committed_offset();
     }
 
+    model::offset get_dirty_offset() const { return _raft0->dirty_offset(); }
+
+    static const bytes invariants_key;
```

Comment:
Why is this needed?

Reply:
it is used in `admin_server` in controller status api